db/compaction_iterator.cc

```diff
@@ -662,4 +655,28 @@ inline bool CompactionIterator::ikeyNotNeededForIncrementalSnapshot() {
          (ikey_.sequence < preserve_deletes_seqnum_);
 }
 
+bool CompactionIterator::IsInEarliestSnapshot(SequenceNumber sequence) {
+  assert(snapshot_checker_ != nullptr);
+  auto in_snapshot =
+      snapshot_checker_->CheckInSnapshot(sequence, earliest_snapshot_);
+  auto snapshots_iter = snapshots_->end();
+  while (UNLIKELY(in_snapshot == SnapshotCheckerResult::kSnapshotReleased)) {
+    if (snapshots_iter == snapshots_->end()) {
+      snapshots_iter = std::lower_bound(snapshots_->begin(), snapshots_->end(),
```

Comment:
can we maintain an earliest_snapshot_iter_ pointer? It would be more efficient and also makes the code easier to read.

Reply:
great suggestion.